hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/validate/HazelcastSqlValidator.java

```diff
@@ -422,6 +430,26 @@ public void validateCall(SqlCall call, SqlValidatorScope scope) {
         super.validateCall(call, scope);
     }
 
+    @Override
+    protected void validateTableFunction(SqlCall node, SqlValidatorScope scope, RelDataType targetRowType) {
+        if (ssc.isSecurityEnabled() && node instanceof SqlBasicCall && !node.getOperandList().isEmpty()) {
+            SqlNode sqlNode = node.getOperandList().get(0);
+            if (sqlNode instanceof SqlBasicCall) {
+                SqlBasicCall call = (SqlBasicCall) sqlNode;
+                SqlOperator operator = call.getOperator();
+                if (operator instanceof HazelcastDynamicTableFunction) {
+                    HazelcastDynamicTableFunction f = (HazelcastDynamicTableFunction) operator;
+                    for (Permission permission : f.permissions(call)) {
+                        ssc.checkPermission(permission);
+                    }
+                }
+            }
+        }
+
+        super.validateTableFunction(node, scope, targetRowType);
+    }
```

Comment:
Question from Calcite noob: why do we do the permission check here, where we don't have the dynamic parameters resolved, and not at some later phase, where we have it?

The `FileSqlConnector.resolveAndValidateFields` is called from here https://github.com/hazelcast/hazelcast/blob/fa71c05f46a91ce809cbc119e2f50f734c880fbc/hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/file/FileTableFunction.java#L59-L63

which is called from here

https://github.com/hazelcast/hazelcast/blob/fa71c05f46a91ce809cbc119e2f50f734c880fbc/hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/schema/HazelcastDynamicTableFunction.java#L84-L87

Is it too difficult to get the `SqlSecurityContext` there? Maybe this choice should be described in the PR description.

Reply:
https://hazelcast.slack.com/archives/D01ST3W9UEN/p1698048821785729